src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundle.java

```diff
@@ -111,6 +113,9 @@ public void initialize(Bootstrap<?> bootstrap) {
 
     @Override
     public void run(T configuration, Environment environment) throws Exception {
+        val portSchemeResolver = createPortSchemeResolver();
+        Preconditions.checkNotNull(portSchemeResolver, "Port scheme resolver can't be null");
+        val portScheme = portSchemeResolver.resolve(configuration.getServerFactory());
```

Comment:
Pass the whole configuration instead of server factory for flexibility. Folks might be starting other servers here and might want to make those discoverable as well